front/src/applications/editor/tools/rangeEdition/utils.ts

```diff
@@ -383,11 +387,11 @@ export const makeSpeedRestrictionTrackRanges = (
     zoneTrackRanges.push(extraTrackRange);
     returnedExtra = true;
   }
-  const trackRangesWithRenamedDirections = zoneTrackRanges.map((tr) => ({
+  const trackRangesWithBothDirections = zoneTrackRanges.map((tr) => ({
     ...tr,
     applicable_directions: 'BOTH' as ApplicableDirections,
   }));
-  return [trackRangesWithRenamedDirections, returnedExtra];
+  return [trackRangesWithBothDirections, returnedExtra];
```

Comment:
Don't use an array but an object. Relying on the order of the objects in the array is less robust than using an object directly
```suggestion
  return {trackRangesWithBothDirections, returnedExtra};
```

Reply:
done